### ISSUE TITLE
What's New section is back after todays Steam update, yet again.

### ISSUE THREAD

**Issue by HazeyDays** [zero318/BegoneWhatsNew#9]
What else is new… Well, ramming it down our throat no longer is.

**Comment by HazeyDays** [comment]
Also, they did something with the section that apparently when you have recent uninstalled DLC's, it will show them there instead, or maybe along with. Which might now make it harder to get rid of 'What's New'. #FML

EDIT: So I was wrong. In actuality, now, if you have anything pre-order or pre-download, it will show up there instead.

**Comment by MateusAuri** [comment]
I'm not really sure what I'm doing, but seems like the CSS code to the What's New bar is pointing to `.library_Container_3xRRJ` in `steamui\css\library.css:134`
I'll try messing with it a bit

EDIT: wait, `.libraryhome_UpdatesContainer_17uEB` in `chunk~547eb3232.css:54` is probably more important... adding `display:none` to it makes _What's New_ disappear properly.
EDIT2: yeah, now I modified it in the file instead of the dev element inspector, and it started the client without _What's New_, as expected. It's basically the same workaround that another user had suggested last time, but here it is in baby steps:

1. Open `steamui\css\chunk~547eb3232.css` in notepad++ or whatever
2. Ctrl+F `.libraryhome_UpdatesContainer_17uEB` just once
3. Right next to the first appearance of `.libraryhome_UpdatesContainer_17uEB` you just found, swap `box-sizing:border-box;` for `display:none;         ` (with 9 space characters, to keep the same file size and avoid losing the changes when Steam checks for changes)
4. That's it have fun enjoy

A possible simpler solution is to just put `display:none` in the place of `height:324px` since they have the same number of characters, but I haven't tried that.

**Comment by HazeyDays** [comment]
> I'm not really sure what I'm doing, but seems like the CSS code to the What's New bar is pointing to `.library_Container_3xRRJ` in `steamui\css\library.css:134` I'll try messing with it a bit
> 
> EDIT: wait, `.libraryhome_UpdatesContainer_17uEB` in `chunk~547eb3232.css:54` is probably more important... adding `display:none` to it makes _What's New_ disappear properly. EDIT2: yeah, now I modified it in the file instead of the dev element inspector, and it started the client without _What's New_, as expected. It's basically the same workaround that another user had suggested last time, but here it is in baby steps:
> 
>     1. Open `steamui\css\chunk~547eb3232.css` in notepad++ or whatever
> 
>     2. Ctrl+F `.libraryhome_UpdatesContainer_17uEB` just once
> 
>     3. Right next to the first appearance of `.libraryhome_UpdatesContainer_17uEB` you just found, swap `box-sizing:border-box;` for `display:none;         ` (with 9 space characters, to keep the same file size and avoid losing the changes when Steam checks for changes)
> 
>     4. That's it have fun enjoy
> 
> 
> A possible simpler solution is to just put `display:none` in the place of `height:324px` since they have the same number of characters, but I haven't tried that.

As soon as I opened Notepad++ and opened Find, it still had that saved from when I used it last time, (which I think had me replace the whole thing with spaces though, can't 100% remember).

Anyway, thanks! I appreciate the time you took to check this out.

EDIT: So the first instructions seemed to just get updated by Steam to revert it. Still shows What's New.
         However, the last idea you had (“height:324px” ⇉ “display:none”) seems to do the trick!

**Comment by jamsz** [comment]
> > I'm not really sure what I'm doing, but seems like the CSS code to the What's New bar is pointing to `.library_Container_3xRRJ` in `steamui\css\library.css:134` I'll try messing with it a bit
> > EDIT: wait, `.libraryhome_UpdatesContainer_17uEB` in `chunk~547eb3232.css:54` is probably more important... adding `display:none` to it makes _What's New_ disappear properly. EDIT2: yeah, now I modified it in the file instead of the dev element inspector, and it started the client without _What's New_, as expected. It's basically the same workaround that another user had suggested last time, but here it is in baby steps:
> > ```
> > 1. Open `steamui\css\chunk~547eb3232.css` in notepad++ or whatever
> > 
> > 2. Ctrl+F `.libraryhome_UpdatesContainer_17uEB` just once
> > 
> > 3. Right next to the first appearance of `.libraryhome_UpdatesContainer_17uEB` you just found, swap `box-sizing:border-box;` for `display:none;         ` (with 9 space characters, to keep the same file size and avoid losing the changes when Steam checks for changes)
> > 
> > 4. That's it have fun enjoy
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > A possible simpler solution is to just put `display:none` in the place of `height:324px` since they have the same number of characters, but I haven't tried that.
> 
> As soon as I opened Notepad++ and opened Find, it still had that saved from when I used it last time, (which I think had me replace the whole thing with spaces though, can't 100% remember).
> 
> Anyway, thanks! I appreciate the time you took to check this out.
> 
> EDIT: So the first instructions seemed to just get updated by Steam to revert it. Still shows What's New. However, the last idea you had (“height:324px” ⇉ “display:none”) seems to do the trick!

These work for me for a minute after opening steam then the section reappears.

**Comment by MateusAuri** [comment]
> These work for me for a minute after opening steam then the section reappears.

Did you make sure not to alter the number of characters / the file size? 
Steam will do that if it detects that the file size is not the same as it used to be before the changes.
I went through 2 Steam updates and a restart since altering the file, and the section still hasn't reappeared.
I also have the old BegoneWhatsNew skin selected in my options, but I'm not sure if it changes anything. 🤔

**Comment by jamsz** [comment]
I only changed height:324px to display:none, but the file is actually two bytes bigger... I'm using vim.

**Comment by MateusAuri** [comment]
Maybe you inserted a line break accidentally? 
Sometimes I do that by pressing enter when I didn't need to (in other contexts, not vim), and it seems to increase exactly 2 bytes in a file.

**Comment by HaikenEdge** [comment]
Can confirm the solution presented in the edit works.

**Comment by jamsz** [comment]
> Maybe you inserted a line break accidentally? Sometimes I do that by pressing enter when I didn't need to (in other contexts, not vim), and it seems to increase exactly 2 bytes in a file.

Yep vim was doing it on its own. With `vim -b` then `:set noeol` it works. Thanks.

**Comment by J3ekir** [comment]
Solution works but not effective I think. It shouldn't need a manual update when Steam is updated. Isn't there a way to make this with skins? Doesn't them override things? I couldn't find a nice documentation about skins.

Also there is `-noverifyfiles` and `-norepairfiles` flags for file size. Didn't check but should work.

Right now, wrote  a Python script that edits the .css file. Put it in task scheduler. Runs when I logged in.

**Comment by MateusAuri** [comment]
> Solution works but not effective I think. It shouldn't need a manual update when Steam is updated. Isn't there a way to make this with skins? Doesn't them override things? I couldn't find a nice documentation about skins.
> 
> Also there is `-noverifyfiles` and `-norepairfiles` flags for file size. Didn't check but should work.
> 
> Right now, wrote a Python script that edits the .css file. Put it in task scheduler. Runs when I logged in.

Yeah, it's more of a workaround than a viable solution per se. It used to be easily solvable with a skin (which was the entire point of this repository), but Steam changed the way it works so now the What's New section data lies in a separate file that can't be changed by skins (as far as I know).

**Comment by J3ekir** [comment]
Sad to have "skins" that changes .css but not every .css. Thanks again, Steam.

Is there a way to apply the solution without restarting Steam. Because I couldn't a way to run the Python script before Steam restarts but after updates.

**Comment by HazeyDays** [comment]
No longer works in the new UI Update Beta. The file wasn't changed, so they moved the data somewhere else. Might work after it switches to stable, but probably won’t.

**Comment by drunohazarb** [comment]
Replacing "height:324px" with "display:none" still works in the new Beta, but the modification has now to be done to "sp.css" instead of "chunk~547eb3232.css".

**Comment by jorp** [comment]
> Replacing "height:324px" with "display:none" still works in the new Beta, but the modification has now to be done to "sp.css" instead of "chunk~547eb3232.css".

I can corroborate that this works for me as well on the following steam version on both Fedora 37* and Windows 10:
```
Steam Version: 1682723851
Steam Client Build Date: Fri, Apr 28 19:55 UTC -08:00
Steam Web Build Date: Fri, Apr 28 18:56 UTC -08:00
Steam API Version: SteamClient020
```

\* The Linux client doesn't seem to persist a restart of steam though..

---
For anyone unsure of how to do this, replace the text as quoted above in the following file locations within the `.libraryhome_UpdatesContainer_17uEB` element in the respective paths for the  Linux distribution of your choice and Windows 10 respectively:

`~/.local/share/Steam/steamui/css/sp.css` 

`C:\Program Files (x86)\Steam\steamui\css\sp.css`

**Comment by drunohazarb** [comment]
On Linux you can use this GNU sed oneliner:
```
sed -i 's/.libraryhome_UpdatesContainer_17uEB{box-sizing:border-box;padding:16px 24px 0px 24px;position:relative;height:324px;/.libraryhome_UpdatesContainer_17uEB{box-sizing:border-box;padding:16px 24px 0px 24px;position:relative;display:none;/' ~/.local/share/Steam/steamui/css/sp.css
```

**Comment by HazeyDays** [comment]
> Replacing "height:324px" with "display:none" still works in the new Beta, but the modification has now to be done to "sp.css" instead of "chunk~547eb3232.css".

Went back to stable, but thanks a bunch! I'll keep it in mind when they change to the new UI permanently.
I tried to figure out where it moved to but had no clue where it would of gone.
When I can't figure it out, I come here! :)

How did you know to look in sp.css..?

**Comment by MateusAuri** [comment]
> > Replacing "height:324px" with "display:none" still works in the new Beta, but the modification has now to be done to "sp.css" instead of "chunk~547eb3232.css".
> 
> Went back to stable, but thanks a bunch! I'll keep it in mind when they change to the new UI permanently. I tried to figure out where it moved to but had no clue where it would of gone. When I can't figure it out, I come here! :)
> 
> How did you know to look in sp.css..?

If you start the Steam client in dev mode, you can use the chromium element inspector to look into the HTML, CSS and whatnot
https://developer.valvesoftware.com/wiki/Command_Line_Options#Steam

When you have an HTML element selected, the panel on the right column tells you what CSS code is affecting it and where it comes from

**Comment by jorp** [comment]
Looks like the beta updated to the below version. I can confirm that the above workaround still works.

```
Steam Version: 1683078372
Steam Client Build Date: Tue, May 2 22:27 UTC -08:00
Steam Web Build Date: Tue, May 2 21:22 UTC -08:00
Steam API Version: SteamClient020
```

**Comment by jorp** [comment]
> On Linux you can use this GNU sed oneliner:
> 
> ```
> sed -i 's/.libraryhome_UpdatesContainer_17uEB{box-sizing:border-box;padding:16px 24px 0px 24px;position:relative;height:324px;/.libraryhome_UpdatesContainer_17uEB{box-sizing:border-box;padding:16px 24px 0px 24px;position:relative;display:none;/' ~/.local/share/Steam/steamui/css/sp.css
> ```

I'm not a PowerShell/Windows guy, but this should do the equivalent:
```
(Get-Content 'C:\Program Files (x86)\Steam\steamui\css\sp.css') -Replace '.libraryhome_UpdatesContainer_17uEB{box-sizing:border-box;padding:16px 24px 0px 24px;position:relative;height:324px;', '.libraryhome_UpdatesContainer_17uEB{box-sizing:border-box;padding:16px 24px 0px 24px;position:relative;display:none;/' | Set-Content 'C:\Program Files (x86)\Steam\steamui\css\sp.css'
```

However, after making those changing and relaunching steam, you'll be greeted with a pop dialog box stating that steam extracting an update. This appears to revert the change you just made, causing What's New to reappear.

The tested version is as follows (extremely recent..):

```
Steam version: 1685483787
Steam Client Build Date: Tue, May 30 18:40 UTC -08:00
Steam Web Build Date: Tue, May 30 16:40 UTC -08:00
Steam API Version: SteamClient020
```

**Comment by HazeyDays** [comment]
> Replacing "height:324px" with "display:none" still works in the new Beta, but the modification has now to be done to "sp.css" instead of "chunk~547eb3232.css".

The new UI update is now stable and default. Can confirm this is the new method to stop the "What's New" section from displaying.

To reiterate along with the quote:
"[Path to Steam folder]\steamui\css\cp.css" Search for .libraryhome_UpdatesContainer_17uEB, then above.

**Comment by MateusAuri** [comment]
just in case someone stumbles into here again, they changed it back from `sp.css` to `chunk~2dcc5aaf7.css`

lately I've been just opening the file in notepad++, hitting ctrl+H and replacing
`height:324px`
with
`display:none`

it affects one other element besides "What's New", but so far I haven't missed anything important

**Comment by jorp** [comment]
> just in case someone stumbles into here again, they changed it back from `sp.css` to `chunk~2dcc5aaf7.css`
> 
> lately I've been just opening the file in notepad++, hitting ctrl+H and replacing `height:324px` with `display:none`
> 
> it affects one other element besides "What's New", but so far I haven't missed anything important

Yes, thanks this seems to current location. If you update such in  https://github.com/zero318/BegoneWhatsNew/issues/9#issuecomment-1569237423, both commands should work.

However, as previously mentioned, Steam then triggers an update next launch. Sadly, the "What's New" section reappears shortly after. Looks like there's likely a checksum or something that reverts modified files.

**Comment by MateusAuri** [comment]
> However, as previously mentioned, Steam then triggers an update next launch. Sadly, the "What's New" section reappears shortly after. Looks like there's likely a checksum or something that reverts modified files.

I just got home, opened Steam, waited for it to update, and my WN section is still disabled.
AFAIK it only checks the file size, so it will only revert your changes if you insert or remove too many characters from the css file. 
Or if an actual update actively needs to change something within that file, obviously. Otherwise, just be wary of keeping filesize/character count as is, then it will stay as you left it.

Unless they changed how it works in beta, I'm not in beta so I wouldn't know about that.

**Comment by MateusAuri** [comment]
Update: I've made a script to automate the process
https://github.com/MateusAuri/SteamWhatsNewYeeter
I'll try to keep it updated whenever they change the files around, but no promises...

**Comment by bakkerthehacker** [comment]
I have also made a python script to update the steam css removing What's New section:
https://github.com/bakkerthehacker/remove-whats-new

**Comment by BirMa** [comment]
Also made a "one-liner" that does it. Sadly the class name (the string `_17uEBe5Ri8TMsnfELvs8-N`) changes every time Valve re-compiles their frontend.
So you'll have to start steam in dev mode (`steam -console -dev -tcp`) and press F12 to find the class name after every update.
@Valve Plz configure your webpack (or equivalent) to generate deterministic class name :3
Edit: I think they heard us. They haven't changed that Id in ages now. Praise to GabeN and his crew.

```sh
cd ~/.local/share/Steam/steamui/css && perl -i -pe 'if (s/(\._17uEBe5Ri8TMsnfELvs8-N)\{([^}]+)\}/"$1" . "{display:none !important; " . (" " x (length($2)-length("display:none !important;")-1)) . "}"/e) { print STDERR "Replaced at line $.\n" }' ./chunk~2dcc5aaf7.css
```
